internal/service/networkmanager/vpn_attachment_test.go

```diff
@@ -0,0 +1,328 @@
+				),
+			},
+			{
+				Config: testAccVpnAttachmentConfig_tags2(rName, vpnIP, "segment", "shared", "Name", "test", bgpASN),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVpnAttachmentExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.segment", "shared"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Name", "test"),
+				),
+			},
+			{
+				Config: testAccVpnAttachmentConfig_tags1(rName, vpnIP, "segment", "shared", bgpASN),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVpnAttachmentExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.segment", "shared"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckVpnAttachmentExists(n string, v *networkmanager.SiteToSiteVpnAttachment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Network Manager VPN Attachment ID is set")
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).NetworkManagerConn
+
+		output, err := tfnetworkmanager.FindVpnAttachmentByID(context.Background(), conn, rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		*v = *output
+
+		return nil
+	}
+}
+
+func testAccCheckVpnAttachmentDestroy(s *terraform.State) error {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).NetworkManagerConn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_networkmanager_vpn_attachment" {
+			continue
+		}
+
+		_, err := tfnetworkmanager.FindVpnAttachmentByID(context.Background(), conn, rs.Primary.ID)
+
+		if tfresource.NotFound(err) {
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		return fmt.Errorf("Network Manager VPN Attachment %s still exists", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccVpnAttachmentConfig_base(rName string, bgpASN int, vpnIP string) string {
+	return acctest.ConfigCompose(acctest.ConfigAvailableAZsNoOptIn(), fmt.Sprintf(`
+data "aws_region" "current" {}
+
+resource "aws_customer_gateway" "test" {
+  bgp_asn    = %[2]d
+  ip_address = %[3]q
+  type       = "ipsec.1"
+}
+
+resource "aws_vpn_connection" "test" {
+  customer_gateway_id = aws_customer_gateway.test.id
+  type                = "ipsec.1"
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_networkmanager_global_network" "test" {
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "awscc_networkmanager_core_network" "test" {
+  global_network_id = aws_networkmanager_global_network.test.id
+  policy_document   = jsonencode(jsondecode(data.aws_networkmanager_core_network_policy_document.test.json))
```

Comment:
Is `jsonencode(jsondecode(...))` needed?

Reply:
Unfortunately, yes 😞.